### PR TITLE
vfat: Add check for invalid characters in bd_fs_vfat_check_label()

### PR DESCRIPTION
**Pull request by tbzatek**
Backported from UDisks: udiskslinuxfilesystem.c:handle_set_label()

### CODE REVIEW

File: src/plugins/fs/vfat.c [hunk 1]
```diff
@@ -288,12 +288,26 @@ gboolean bd_fs_vfat_set_label (const gchar *device, const gchar *label, GError *
...
+
     if (strlen (label) > 11) {
         g_set_error (error, BD_FS_ERROR, BD_FS_ERROR_LABEL_INVALID,
                      "Label for VFAT filesystem must be at most 11 characters long.");
         return FALSE;
     }
 
+    /* VFAT does not allow some characters; as dosfslabel does not enforce this,
+     * check in advance; also, VFAT only knows upper-case characters, dosfslabel
+     * enforces this */
```
Comment (vojtechtrefny) [inline src/plugins/fs/vfat.c:302]: > VFAT only knows upper-case characters, dosfslabel enforces this

`dosfslabel` only warns about it. Maybe we should be stricter here and force uppercase only?

```
$ sudo dosfslabel /dev/loop1 aaaa
fatlabel: warning - lowercase labels might not work properly on some systems
```

**Review by vojtechtrefny — APPROVED** [review]
Looks good. I took the label "limits" from blivet, maybe we should check all filesystems to be sure these are actually valid.

File: src/plugins/fs/vfat.c (hunk 1 above)
Comment (tbzatek) [reply src/plugins/fs/vfat.c:302]: Right, though I'm not sure if it's worth enforcing that - i.e. failing the check. UDisks always converts FAT labels uppercase without complaining.

Pushed a second commit that always converts labels uppercase, please check. The `bd_fs_vfat_mkfs()` function still allows specifying arbitrary arguments, added a docs note.

File: src/plugins/fs/vfat.c [hunk 2]
```diff
@@ -274,7 +282,14 @@ gboolean bd_fs_vfat_set_label (const gchar *device, const gchar *label, GError *
             args[2] = "--reset";
     }
 
-    return bd_utils_exec_and_report_error (args, NULL, error);
+    /* forcefully convert the label uppercase */
+    if (!new_vfat && label)
```
Comment (vojtechtrefny) [inline src/plugins/fs/vfat.c:286]: Why only for dosfstools < 4.2? You convert it for both the old and new version in `bd_fs_vfat_mkfs_options`.

Comment (tbzatek) [reply src/plugins/fs/vfat.c:286]: That's a logical conundrum. `new_vfat` is only set when dosfstools >= 4.2 AND the label is empty.

Comment (vojtechtrefny) [reply src/plugins/fs/vfat.c:286]: Oh, I see. But I think adding a new `is_empty`  boolean would make this much easier to read.

Comment (tbzatek) [reply src/plugins/fs/vfat.c]: Attempt #2, please check.

**Review by vojtechtrefny — APPROVED** [review]
Looks good, thanks.

**Pull request merged**